crates/shared/src/sources/swapr/reader.rs

```diff
@@ -0,0 +1,134 @@
+//! A pool state reading implementation specific to Swapr.
+
+use crate::{
+    sources::uniswap_v2::pool_fetching::{self, DefaultPoolReader, Pool, PoolReading},
+    Web3CallBatch,
+};
+use anyhow::Result;
+use contracts::ISwaprPair;
+use ethcontract::{errors::MethodError, BlockId};
+use futures::{future::BoxFuture, FutureExt as _};
+use model::TokenPair;
+use num::rational::Ratio;
+
+/// A specialized Uniswap-like pool reader for DXdao Swapr pools.
+///
+/// Specifically, Swapr pools have dynamic fees that need to be fetched with the
+/// pool state.
+pub struct SwaprPoolReader(DefaultPoolReader);
```

Comment:
So the default pool reader is the one which also fetches fees. Makes sense because uniswap fees are permanently fixed to be 0.3%. Is that true? Would it be possible for the uni holders to vote to change the fees for a specific pool? I haven't actually checked, just wondering.

Reply:
> So the default pool reader is the one which also fetches fees

No, only the `SwaprPoolReader` does. The default one doesn't have to because Uniswap pools have a fixed 0.3% fee.

> Would it be possible for the uni holders to vote to change the fees for a specific pool?

No, its a hard-coded value: 

https://github.com/Uniswap/v2-core/blob/4dd59067c76dea4a0e8e4bfdda41877a6b16dedc/contracts/UniswapV2Pair.sol#L180-L181